src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -141,6 +142,7 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
                 restoreArgs.Log);
             
             var rootPath = Path.GetDirectoryName(project.PackageSpec.FilePath);
+            var revocationMode = SettingsUtility.GetRevocationMode(settings);
```

Comment:
This won't work very well.

You can have different settings objects for different projects, which means depending on which project's restore won, you might get a different revocation mode check.

Reply:
I think it's fine to have different revocation mode for different projects if customer has configured his config files that way... that's just my opinion!

Reply:
I disagree. 

Say both projA and projB reference the same version of PackageX. 

One says offline revocation the other says online. 

Say the certificate has been revoked. 

What happens in that situation? Do you install the package? If you do, both will use it. 
I'm not sure even sure how the deduplication happens in our code, but we might end up favoring one over the other. 

Later we will doing the same thing for client policies and trusted signers and get in an even shittier situation.

Reply:
`What happens in that situation? Do you install the package?`

AFAIK irrespective of the fact that revocation mode is online or offline, package gets installed since this is not a hard failure. It's just the matter of performance of trying to go online when there is no network available. @PatoBeltran can correct me if my understanding is wrong.

Reply:
It's not quite the same, there are subtle differences. 

My issue is that we will apply this same exact pattern when doing client policies.

Reply:
That's right packages do get installed by default today irrespective or revocation check result.

And I agree with @anangaur - "I don’t get why this is different from any other NuGet settings. For example, the sources could also vary so could any other setting if there is nuget.config for each project. I feel the existing solution is good enough. Maybe we need to also need to support it as an ENV var."

Reply:
> And I agree with @anangaur - "I don’t get why this is different from any other NuGet settings. For example, the sources could also vary so could any other setting if there is nuget.config for each project. I feel the existing solution is good enough. Maybe we need to also need to support it as an ENV var."

Because it's a worse contract you're breaking when you potentially allow a package that's does not satisfy the security requirements set by the project.

Even if we don't do anything special, we need to call this out, people need to be aware of the consequences of something like this potentially happening.

Ex: Are we deterministic in this scenario? What will happen when you restore both projects at the same time etc.  
We need to exactly how our product will behave in these scenario so that we minimize the friction when customers do hit these scenarios.

Reply:
I had this discussed offline with @PatoBeltran and it seems this is a legit issue with having multiple config files with different revocation mode at differnt projects. It will create intermittent issues with restore or install.

Then I also discussed with @dtivel about the ENVIRONMENT variable approach instead of a nuget config for this setting and he agreed at this time, we can go ahead with env variable for this issue.

Tomorrow, we anyways have to solve the bigger issue of having some kind of global settings or user level settings which cant be over-written at solution or project level.

Reply:
@PatoBeltran can correct me if I am wrong, but these are per package warnings. We also have a cert  cache. If the cert cache was refreshed because one project is in online mode, revocation check for the offline project would pass since we are able to use the cache.

Let's say the cert cache does not have the cert, the warning would show up for the offline project, but not for the online one.

I feel that's a perfectly reasonable behavior.